gadgets/mri_core/GenericCartesianGrappaReconGadget.cpp

```diff
@@ -0,0 +1,1011 @@
+
+#include "GenericCartesianGrappaReconGadget.h"
+#include <iomanip>
+
+#include "mri_core_kspace_filter.h"
+#include "hoNDArray_reductions.h"
+#include "mri_core_def.h"
+#include "mri_core_coil_map_estimation.h"
+#include "mri_core_grappa.h"
+#include "mri_core_utility.h"
+
+/*
+    The input is IsmrmrdReconData and output is single 2D or 3D ISMRMRD images
+
+    If required, the gfactor map can be sent out
+
+    If the  number of required destination channel is 1, the GrappaONE recon will be performed
+
+    The image number computation logic is implemented in compute_image_number function, which can be overloaded
+*/
+
+namespace Gadgetron {
+
+    GenericCartesianGrappaReconGadget::GenericCartesianGrappaReconGadget() : num_encoding_spaces_(1), process_called_times_(0)
+    {
+        gt_timer_.set_timing_in_destruction(false);
+        gt_timer_local_.set_timing_in_destruction(false);
+
+        std::string procTime;
+        Gadgetron::get_current_moment(procTime);
```

Comment:
get_current_moment should simply return a string, rather than take one as input. Consider calling it something like current_timestring